examples/api/lib/widgets/routes/flexible_route_transitions.0.dart

```diff
@@ -0,0 +1,226 @@
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              TextButton(
+                onPressed: () {
+                  Navigator.of(context).push(
+                    VerticalTransitionPageRoute<void>(
+                      builder: (BuildContext context) {
+                        return  const MyHomePage(title: 'Crazy Vertical Transition');
+                      },
+                    ),
+                  );
+                },
+                child: const Text('Crazy Vertical Transition'),
+              ),
+              TextButton(
+                onPressed: () {
+                  Navigator.of(context).push(
+                    MaterialPageRoute<void>(
+                      builder: (BuildContext context) {
+                        return  const MyHomePage(title: 'Zoom Transition');
+                      },
+                    ),
+                  );
+                },
+                child: const Text('Zoom Transition'),
+              ),
+              TextButton(
+                onPressed: () {
+                  final CupertinoPageRoute<void> route = CupertinoPageRoute<void>(
+                    builder: (BuildContext context) {
+                      return  const MyHomePage(title: 'Cupertino Transition');
+                    }
+                  );
+                  Navigator.of(context).push(route);
+                },
+                child: const Text('Cupertino Transition'),
+              ),
+            ],
+          ),
+        ),
+    );
+  }
+}
+
+class VerticalTransitionPageRoute<T> extends PageRoute<T> {
+  /// Creates a page route for use in an iOS designed app.
+  VerticalTransitionPageRoute({
+    required this.builder,
+  });
+
+  final WidgetBuilder builder;
+
+  @override
+  DelegatedTransitionBuilder? get delegatedTransition => VerticalPageTransition._delegatedTransition;
+
+  @override
+  Color? get barrierColor => const Color(0x00000000);
+
+  @override
+  bool get barrierDismissible => false;
+
+  @override
+  String? get barrierLabel => 'Should be no visible barrier...';
+
+  @override
+  bool get maintainState => true;
+
+  @override
+  bool get opaque => false;
+
+  // Begin PageRoute.
+
+  @override
+  Duration get transitionDuration => const Duration(milliseconds: 2000);
+
+  @override
+  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+    return builder(context);
+  }
+
+  static Widget buildPageTransitions<T>(
+    ModalRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return VerticalPageTransition(
+      primaryRouteAnimation: animation,
+      secondaryRouteAnimation: secondaryAnimation,
+      child: child,
+    );
+  }
+
+  // TODO(justinmc): No canTransitionTo needed? I want to be able to have
+  // back-to-back VerticalPageTransitions.
```

Comment:
Don't forget to remove this. The question stands though. I guess it doesn't need canTransitionTo unless we want to demonstrate that somehow?